CorsixTH/Src/th_lua.cpp

```diff
@@ -185,15 +185,23 @@ int l_get_compile_options(lua_State* L) {
   lua_settop(L, 0);
   lua_newtable(L);
 
+#ifdef __x86_64__
 #ifdef CORSIX_TH_64BIT
-  lua_pushboolean(L, 1);
+  lua_pushliteral(L, "x64");
 #else
-  lua_pushboolean(L, 0);
+  lua_pushliteral(L, "x86");
 #endif
-  lua_setfield(L, -2, "arch_64");
 
-  lua_pushliteral(L, "SDL");
-  lua_setfield(L, -2, "renderer");
+#elif __arm__
+#ifdef CORSIX_TH_64BIT
+  lua_pushliteral(L, "ARM64");
+#else
+  lua_pushliteral(L, "ARM");
+#endif
+
```

Comment:
You are missing an else here, so Other isn't pushed when neither __x86_64__ nor __arm__ are defined.